### PR TITLE
EVP-2011 - add cassandra check to init db

### PR DESCRIPTION
**Pull request by ricardnacher**

### CODE REVIEW

File: helm/thingsboard/templates/initializedb-job.yaml [hunk 1]
```diff
@@ -27,7 +27,14 @@ spec:
       imagePullSecrets:
         - name: '{{ .Release.Name }}-container-registry'
       initContainers:
-        - name: check-db-ready
+        {{- if .Values.cassandra.enabled }}
+        - name: check-cassandra-db-ready
+          image: 'cassandra:3.11'
```
Comment (dmitry-midokura) [inline helm/thingsboard/templates/initializedb-job.yaml:32]: Nit: why is this tag hardcoded here? Can we use same tag as cassandra chart?

Comment (ricardnacher) [reply helm/thingsboard/templates/initializedb-job.yaml:32]: I haven't found a good way for getting this value from another variable in the chart.
I tryied to use the bitnami image but it complains with an error. 
Let's keep it hardcoded for now as it is in thingsboard-private repository.

File: helm/thingsboard/values.yaml [hunk 2]
```diff
@@ -271,6 +271,9 @@ cassandra:
   replicaCount: 1
   dbUser:
     password: setplease
+  extraEnvVars:
+  - name: JVM_OPTS
```
Comment (dmitry-midokura) [inline helm/thingsboard/values.yaml:275]: Why is this needed? These affect both "normal" cassandra and the init job, right?

**Review by cclavero-mk — APPROVED** [review]

File: helm/thingsboard/values.yaml (hunk 2 above)
Comment (ricardnacher) [reply helm/thingsboard/values.yaml:275]: It is needed because nodetool uses JMX for interaction. It affects "normal" cassandra nodes.

**Comment by xavileon** [comment]
This is probaby related to all the PRs to this repo, just to get context and how we operate today:
- how divergent are the upstream and downstream repos? are we planning to contribute, rebase from time to time or keep the fork evolving separately?
- how are all these changes to charts being tested?

**Comment by ricardnacher** [comment]
> This is probaby related to all the PRs to this repo, just to get context and how we operate today:
> 
> * how divergent are the upstream and downstream repos? are we planning to contribute, rebase from time to time or keep the fork evolving separately?
> * how are all these changes to charts being tested?

I am afraid that they are getting more and more divergent. 
We planned to contribute and it is still our goal, but they were not reviewing any PR for a long time. It seems that right now there is activity again in the upstream repository. 

Changes are tested locally.

**Comment by ricardnacher** [comment]
FYI: I created a task for handling this synchronization. https://midokura.atlassian.net/browse/EVP-2126

**Comment by xavileon** [comment]
do you see any value in running some tests on this repo or it's not worth it? e.g. a run of one e2e test (maybe not necessarily the whole suite) with the generated chart

**Comment by xavileon** [comment]
> At some point "ALL" tests of the upstream repository have to be run for , first, being confident with our changes and second, being mergeable to upstream repository. WDYT?

I didn't see any test runs in the [upstream](https://github.com/thingsboard/thingsboard-ce-k8s) repository of this chart (could it be they are in some place I didn't notice). The release of this chart seems untested, could this lack of tests lead to the same issues as with the release of the evp-api chart?

**Comment by ricardnacher** [comment]
> > At some point "ALL" tests of the upstream repository have to be run for , first, being confident with our changes and second, being mergeable to upstream repository. WDYT?
> 
> I didn't see any test runs in the [upstream](https://github.com/thingsboard/thingsboard-ce-k8s) repository of this chart (could it be they are in some place I didn't notice). The release of this chart seems untested, could this lack of tests lead to the same issues as with the release of the evp-api chart?

You are right! There are no tests in the upstream repository :scream: 
Manual tests were done here: https://github.com/midokura/thingsboard-ce-k8s/pull/8
